src/test/kotlin/org/opensearch/replication/bwc/BackwardsCompatibilityIT.kt

```diff
@@ -0,0 +1,94 @@
+package org.opensearch.replication.bwc;
+
+import org.assertj.core.api.Assertions
+import org.junit.Assert
+import org.opensearch.action.DocWriteResponse
+import org.opensearch.action.admin.cluster.health.ClusterHealthRequest
+import org.opensearch.action.get.GetRequest
+import org.opensearch.action.index.IndexRequest
+import org.opensearch.client.RequestOptions
+import org.opensearch.client.indices.CreateIndexRequest
+import org.opensearch.replication.MultiClusterAnnotations
+import org.opensearch.replication.MultiClusterRestTestCase
+import org.opensearch.replication.StartReplicationRequest
+import org.opensearch.replication.startReplication
+import org.opensearch.replication.stopReplication
+import org.opensearch.test.OpenSearchTestCase.assertBusy
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+import java.util.stream.Collectors
+
+
+const val LEADER = "bwcLeader"
+const val FOLLOWER = "bwcFollower"
+
+@MultiClusterAnnotations.ClusterConfigurations(
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${LEADER}0"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${FOLLOWER}0"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${LEADER}1"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${FOLLOWER}1")
+)
+class BackwardsCompatibilityIT : MultiClusterRestTestCase() {
+    private val NUM_NODES = 3
+    private val REPLICATION_PLUGIN_NAME = "opensearch-cross-cluster-replication"
+    private val clusterSuffix = System.getProperty("tests.clustersuffix")
+    private val leader_name = "${LEADER}$clusterSuffix"
+    private val follower_name = "${FOLLOWER}$clusterSuffix"
+    @Throws(Exception::class)
+    fun testReplicationPlugin() {
+        verifyReplicationPluginInstalled(leader_name)
+        verifyReplicationPluginInstalled(follower_name)
+        verifyReplication()
+    }
+
+    private fun verifyReplication() {
+        val follower = getClientForCluster(follower_name)
```

Comment:
Can we start replication on older version and check the status in mixed mode and lastly in upgraded cluster?